### ISSUE TITLE
Converter for bitronvideo AV2010/24A smoke detector

### ISSUE THREAD

**Issue by Mouky66** [Koenkk/zigbee-herdsman-converters#213]
Hi,
is there a converter for bitronvideo smoke detector AV2010/24A?
Thank you

**Comment by Koenkk** [comment]
This device is not supported, please follow https://koenkk.github.io/zigbee2mqtt/how_tos/how_to_support_new_devices.html

**Comment by Mouky66** [comment]
i tried to, but all i get is:
`zigbee-shepherd-converters:devices Configured '(cb) => device.write('ssIasZone', 'iasCieAddr', coordinator.device.getIeeeAddr(), cb)' with result 'Error: AF data request fails, status code: 240. MAC transaction expired.' +0ms
  zigbee2mqtt:error 2019-1-15 18:36:49 Failed to configure Rauchmelder 0x00124b00091604fc`

with config:
`{ 
      zigbeeModel: ['902010/24A'], 
      model: '902010/24A',
      vendor: 'Bitronvideo', 
      description: '',
      supports: '', 
      fromZigbee: [fz.generic_battery],
      toZigbee: [],
      configure: (ieeeAddr, shepherd, coordinator, callback) => {
           const device = shepherd.find(ieeeAddr, 1);
           const actions = [
                  (cb) => device.write('ssIasZone', 'iasCieAddr', coordinator.device.getIeeeAddr(), cb),
                  (cb) => device.write('ssIasZone', 'iasCieAddr', coordinator.device.getIeeeAddr(), cb),
                  (cb) => device.report('ssIasZone', 'zoneStatus', 0, 30, null, cb),
                  (cb) => device.report('ssIasZone', 'zoneStatus', 0, 30, null, cb),
                  (cb) => device.functional('ssIasZone', 'enrollRsp', {enrollrspcode: 1, zoneid: 50}, cb),
                  (cb) => device.functional('ssIasZone', 'enrollRsp', {enrollrspcode: 1, zoneid: 50}, cb),
              ];
               execute(device, actions, callback);
  
          }, 
    },`

**Comment by Koenkk** [comment]
Could the device be asleep? You need to wake it up when starting zigbee2mqtt.

**Comment by Koenkk** [comment]
Configure only has to succeed once, meaning your device is setup properly now.

**Comment by SneakyV** [comment]
@Mouky66 
I also have a Bitron smoke detector and wanted to ask if you can give me the finished driver for zigbee2mqtt for the 902010 / 24A?

That would be really nice :-)

**Comment by Mouky66** [comment]
I am sorry, but i have not finished my setup yet. I have connected the detector via raspbee (deconz) to my home-assistant and there it works fine. I will continue trying in a few weeks.

**Comment by SneakyV** [comment]
Thank you for the feedback. I would be glad, if you post the configuration here later. 

Vielen Dank für die Rückmeldung. Ich würde mich freuen, wenn Sie später die Konfiguration hier posten würden. Ich werde die Einrichtung aber auch selbst versuchen.

**Comment by chb383** [comment]
I would like to step in here. I have a bunch of these smoke detectors I like to connect. 
I´m not really good in coding especially not js. But I tried to follow the "how to support new devices". 
And now my device seems to connect, but I do not get any WARNs or other reaction from the device.

Thats a part of my log after connecting the device:
`
2019-05-22 20:23:19.909 - error: zigbee.0 Unknown device model 902010/24A emit event readRsp with data:{"cid":"genBasic","data":{"zclVersion":1,"appVersion":20,"hwVersion":0,"manufacturerName":"Bitron Video","modelId":"902010/24A"}}
2019-05-22 20:23:20.559 - error: zigbee.0 Unknown device model 902010/24A emit event readRsp with data:{"cid":"genBasic","data":{"dateCode":"20160715 ","powerSource":3,"locationDesc":" ","physicalEnv":0,"deviceEnabled":1}}
2019-05-22 20:23:20.651 - error: zigbee.0 Unknown device model 902010/24A emit event readRsp with data:{"cid":"genBasic","data":{"dateCode":"20160715 ","powerSource":3,"locationDesc":" ","physicalEnv":0,"deviceEnabled":1}}
2019-05-22 20:23:21.927 - error: zigbee.0 Unknown device model 902010/24A emit event readRsp with data:{"cid":"genIdentify","data":{"identifyTime":0}}
2019-05-22 20:23:23.002 - error: zigbee.0 Unknown device model 902010/24A emit event readRsp with data:{"cid":"ssIasZone","data":{"zoneState":0,"zoneType":40,"zoneStatus":16,"iasCieAddr":"0x0000000000000000","zoneId":255}}
2019-05-22 20:23:23.172 - error: zigbee.0 Unknown device model 902010/24A emit event readRsp with data:{"cid":"ssIasZone","data":{"zoneState":0,"zoneType":40,"zoneStatus":16,"iasCieAddr":"0x0000000000000000","zoneId":255}}
2019-05-22 20:23:24.109 - error: zigbee.0 Unknown device model 902010/24A emit event readRsp with data:{"cid":"ssIasWd","data":{"maxDuration":180}}
2019-05-22 20:23:25.589 - error: zigbee.0 Unknown device model 902010/24A emit event readRsp with data:{"cid":"genOta","data":{"upgradeServerId":"0xffffffffffffffff","fileOffset":4294967295,"currentFileVersion":20,"currentZigbeeStackVersion":2,"downloadedFileVersion":4294967295}}
2019-05-22 20:23:26.250 - error: zigbee.0 Unknown device model 902010/24A emit event readRsp with data:{"cid":"genOta","data":{"downloadedZigbeeStackVersion":65535,"imageUpgradeStatus":0,"manufacturerId":0,"imageTypeId":0,"minimumBlockReqDelay":0}}
2019-05-22 20:23:26.377 - error: zigbee.0 Unknown device model 902010/24A emit event readRsp with data:{"cid":"genOta","data":{"70":4294967295,"imageStamp":0}}
2019-05-22 20:23:26.401 - info: zigbee.0 new dev 0x00124b00096a99b2 21570 902010/24A
`
and thats what I made out of it:
`
 {
zigbeeModel: ['902010/24A'],
model: 'AV2010/24A',
vendor: 'Bitron',
description: 'Optical smoke detector',
supports: 'smoke',
fromZigbee: [],
toZigbee: [],
const device = shepherd.find(ieeeAddr, 1);
const actions = [
(cb) => device.bind('genOta', coordinator, cb),
(cb) => device.bind('ssIasWd', coordinator, cb),
(cb) => device.bind('ssIasZone', coordinator, cb),
(cb) => device.bind('genIdentify', coordinator, cb),
(cb) => device.bind('genBasic', coordinator, cb),
(cb) => device.report('ssIasZone', 'zoneStatus', 0, 30, null, cb), //Werte ok??
(cb) => device.functional('ssIasZone', 'enrollRsp', {enrollrspcode: 1, zoneid: 23}, cb), //Werte ok??
 ];
execute(device, actions, callback);
 },
 },
`
Unfortunately I do not really understand how to work with the actions. I have seen @Mouky66 added two lines for each action. Is that right/necessary?
After restarting the adapter, there are no more messages from the device in the log.

Thats what happens after restart:
`
2019-05-22 20:52:24.235 - info: zigbee.0 Reset coordinator
2019-05-22 20:52:25.255 - info: zigbee.0 zigbee-shepherd started!
2019-05-22 20:52:25.271 - info: zigbee.0 zigbee-shepherd ready. version: 2.6.3 rev 20181024
2019-05-22 20:52:25.324 - info: zigbee.0 1 (0x84182600001016cb): AB3257001NJ - OSRAM Smart+ plug (Router)
2019-05-22 20:52:25.327 - info: zigbee.0 2 (0x84182600001054ec): AB3257001NJ - OSRAM Smart+ plug (Router)
2019-05-22 20:52:25.328 - info: zigbee.0 3 (0x00124b0016477368): AV2010/25 - Bitron Video wireless socket (Router)
2019-05-22 20:52:25.329 - info: zigbee.0 4 (0x00124b0002f4fee6): AV2010/25 - Bitron Video wireless socket (Router)
2019-05-22 20:52:25.329 - info: zigbee.0 5 (0x00124b00096a99b2): unkown - undefined unknown (EndDevice)
2019-05-22 20:52:56.196 - info: zigbee.0 Successfully configured 0x84182600001054ec Plug 01
2019-05-22 20:52:56.251 - info: zigbee.0 Successfully configured 0x84182600001016cb Plug 01
2019-05-22 20:52:56.520 - info: zigbee.0 Successfully configured 0x00124b0016477368 902010/25
2019-05-22 20:53:27.249 - info: zigbee.0 new dev 0x84182600001054ec 59251 Plug 01
2019-05-22 20:53:28.186 - info: zigbee.0 Successfully configured 0x84182600001054ec Plug 01
2019-05-22 20:53:28.231 - info: zigbee.0 Successfully configured 0x00124b0002f4fee6 902010/25
2019-05-22 20:53:29.339 - info: zigbee.0 new dev 0x84182600001016cb 47161 Plug 01
2019-05-22 20:53:30.299 - info: zigbee.0 Successfully configured 0x84182600001016cb Plug 01
2019-05-22 20:53:31.070 - info: zigbee.0 new dev 0x00124b0002f4fee6 42727 902010/25
2019-05-22 20:53:31.951 - info: zigbee.0 Successfully configured 0x00124b0002f4fee6 902010/25
2019-05-22 20:53:33.840 - info: zigbee.0 new dev 0x00124b0016477368 41633 902010/25
2019-05-22 20:53:34.792 - info: zigbee.0 Successfully configured 0x00124b0016477368 902010/25
`

Any ideas, support and help appreciated. 

Thank you very much, 
Christian
